### ISSUE TITLE
Type completeness check with generic type aliases

### ISSUE THREAD

**Issue by tkukushkin** [microsoft/pyright#2625]
**To Reproduce**

```python
from typing import TypeVar, List

T = TypeVar('T')
Foo = List[T]

def foo(f: Foo) -> None:  # no complaints
    pass

def bar(f: List) -> None:  # Type of parameter "f" is partially unknown
    pass
```

**Expected behavior**
Pyright should complain about type of parameter `f` for `foo` function too.

**VS Code extension or command-line**
```bash
❯ pyright --version                          
pyright 1.1.190
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release. Here's the commit: https://github.com/microsoft/pyright/commit/880efb3ca773527168f3bbbd5484f329a4e39732.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.191, which I just published. It will also be included in the next release of pylance.

**Comment by tkukushkin** [comment]
I updated pyright, but I can still reproduce the problem.

```
❯ pyright --version
pyright 1.1.191
```

**Comment by erictraut** [comment]
I copied the sample code above into a "py.typed" source file and ran "pyright --verifytypes" on it. It correctly reports the error, whereas it didn't report an error prior to my recent fix. 

Please double check that it's not working for you. If not, I'd appreciate it if you could provide specific repro steps.

**Comment by tkukushkin** [comment]
I created repo https://github.com/tkukushkin/pyright-2625 with this sample code and Dockerfile to run pyright in a clean environment.

My output:
```
❯ docker run -it --rm $(docker build -q .)
pyright 1.1.191
Package name: "pyright_2625"
Package directory: "/app/src/pyright_2625"
Path of py.typed file: "/app/src/pyright_2625/py.typed"

Public modules: 1
   pyright_2625

Symbols used in public interface:
pyright_2625.bar
  /app/src/pyright_2625/__init__.py:9:5 - error: Type of parameter "f" is partially unknown
    Parameter type is "List[Unknown]"
      Type argument 0 has unknown type

Symbols exported by "pyright_2625": 4
  With known type: 3
  With partially unknown type: 1
  Functions without docstring: 2
  Functions without default param: 0
  Classes without docstring: 0

Other symbols referenced but not exported by "pyright_2625": 0
  With known type: 0
  With partially unknown type: 0

Type completeness score: 75%

Completed in 1.042sec
```

Pyright reports one error, but should report two.

**Comment by erictraut** [comment]
Thanks for the details. I was able to repro the problem and fix it. The code was missing a check for a generic type alias with missing type arguments.

**Comment by tkukushkin** [comment]
I added third example in repository:

```python
from typing import TypeVar, List, Union

T = TypeVar('T')
U = Union[str, List[T]]

def baz(x: U) -> None:
    pass
```

Can you check it too with a future fix?

**Comment by erictraut** [comment]
Yes, this is covered by the generic type alias logic that I just added:

```
Symbols used in public interface:
pyright_2625.baz
  pyright-2625/src/pyright_2625/__init__.py:7:5 - error: Type of parameter "x" is partially unknown
    Parameter type is "U[Unknown]"
      Type argument 1 for type alias "U" has unknown type

Symbols exported by "pyright_2625": 3
  With known type: 2
  With partially unknown type: 1
  Functions without docstring: 1
  Functions without default param: 0
  Classes without docstring: 0
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.192, which I just published.
